peru/resources/plugins/git/git_plugin.py

```diff
@@ -167,10 +168,33 @@ def plugin_reup(url, reup):
         print('rev:', output.strip(), file=out_file)
 
 
+def git_default_branch(url) -> str:
+    """
+    With a list of common default branches names, it checks against
+    the available branch list if this possible default branch exists.
+    By default git used to take master as default branch, but newer versions
+    tend to use main, even trunk or development can be found.
+
+    Args:
+        url (str): url from the target repository to be checked.
+    Returns:
+        str: returns a possible match for the git default branch.
+    """
+    repo_path = clone_if_needed(url)
+    default_branches_list = ['master', 'main', 'trunk', 'development']
+    for branch in default_branches_list:
+        output = git(
+            'branch', '-a', git_dir=repo_path, capture_output=True).output
+        parse_output = re.search(r'' + re.escape(branch) + '', output)
+        if parse_output:
```

Comment:
Rather than parsing the text output of `git branch` (which is human-oriented text that might not be perfectly stable), what I'd rather use here is the method suggested at https://stackoverflow.com/questions/5167957/is-there-a-better-way-to-find-out-if-a-local-git-branch-exists. That command has no output, and instead you look at its error code. (`0` means success, i.e. the branch exists. Anything else means failure, i.e. it doesn't exist.) You'll need to use the `checked=False` keyword arg to avoid throwing an exception in that failure case; see the similar examples in the `already_has_ref()` function.

Reply:
Ahhhha, that was the code I was looking for. 
Everytime I was using the git wrapper I had this problem. For this reason I haven't gone this route. Let me see what I can come up.

Reply:
It looks better now